coordinator/statement_executor.go

```diff
@@ -958,13 +954,18 @@ func (e *StatementExecutor) executeShowTagKeys(q *influxql.ShowTagKeysStatement,
 		return err
 	}
 
-	sgis, err := e.MetaClient.ShardGroupsByTimeRange(di.Name, di.DefaultRetentionPolicy, timeRange.MinTime(), timeRange.MaxTime())
-	if err != nil {
-		return err
+	// Get all shards for all retention policies.
+	var allgroups []meta.ShardGroupInfo
+	for _, rpi := range di.RetentionPolicies {
+		sgis, err := e.MetaClient.ShardGroupsByTimeRange(q.Database, rpi.Name, timeRange.MinTime(), timeRange.MaxTime())
+		if err != nil {
+			return err
+		}
+		allgroups = append(allgroups, sgis...)
 	}
```

Comment:
I'm not sure I can stand for the all lowercase variable names. It causes a facial tic just looking at it.

Reply:
haha sorry about that.